storage/innobase/log/log0recv.cc

```diff
@@ -4122,7 +4122,7 @@ static bool recv_scan_log(bool last_phase)
           ut_ad(!end || end == recv_sys.lsn);
           mysql_mutex_unlock(&recv_sys.mutex);
 
-          if (!end)
+          if (!recv_sys.is_corrupt_fs() && !end)
           {
             recv_sys.set_corrupt_log();
```

Comment:
`recv_sys.found_corrupt_fs` is supposed to be protected by `recv_sys.mutex` if you look at `recv_sys_t::set_corrupt_fs()`. I think that we’d better not release `recv_sys.mutex` before we have decided whether to set `recv_sys.found_corrupt_log`.

Another thing is that we’d better check something that likely to reside in a register, before reading something from memory, that is, `!end && recv_sys.is_corrupt_fs()`.

Reply:
Makes sense. Modified the code.